components/script/dom/readablestreamdefaultreader.rs

```diff
@@ -169,45 +165,81 @@ impl ReadableStreamDefaultReader {
             .pop_front()
             .expect("Reader must have read request when remove is called into.")
     }
+
+    fn get_read_requests(&self) -> VecDeque<ReadRequest> {
+        mem::take(&mut *self.read_requests.borrow_mut())
+    }
 }
 
 impl ReadableStreamDefaultReaderMethods for ReadableStreamDefaultReader {
     /// <https://streams.spec.whatwg.org/#default-reader-read>
     fn Read(&self) -> Rc<Promise> {
         let promise = Promise::new(&self.reflector_.global());
 
-        self.stream
-            .perform_pull_steps(ReadRequest::Read(promise.clone()));
+        if let Some(stream) = self.stream.get() {
+            stream.perform_pull_steps(ReadRequest::Read(promise.clone()));
```

Comment:
This should call into https://streams.spec.whatwg.org/#readable-stream-default-reader-read (we don't have that method yet).

Reply:
I needed to changed this code because I converted stream to `MutNullableDom<ReadableStream>`, this `read` function is not implemented correctly, I think we can fix it in next PR when we add https://streams.spec.whatwg.org/#readable-stream-default-reader-read